src/plugins/enzymes/src/FindEnzymesDialog.cpp

```diff
@@ -589,6 +640,58 @@ void FindEnzymesDialog::accept() {
     QDialog::accept();
 }
 
+void FindEnzymesDialog::sl_handleSupplierSelectionChange(QStringList checkedSuppliers) {
+    const auto enzymes = EnzymesSelectorWidget::getLoadedEnzymes();
+    QList<SEnzymeData> visibleEnzymes;
+    for (const auto& enzyme : qAsConst(enzymes)) {
+        for (const auto& supplier : qAsConst(enzyme->suppliers)) {
+            CHECK_CONTINUE(checkedSuppliers.contains(supplier));
+
+            visibleEnzymes.append(enzyme);
+            break;
+
+        }
+    }
+    enzSel->setEnzymesList(visibleEnzymes);
+    static const QString notDefinedTr = EnzymesIO::tr(EnzymesIO::NOT_DEFINED_SIGN);
+    if (checkedSuppliers.contains(notDefinedTr)) {
+        checkedSuppliers.replace(checkedSuppliers.indexOf(notDefinedTr), EnzymesIO::NOT_DEFINED_SIGN);
+    }
+    auto value = checkedSuppliers.join(SUPPLIERS_LIST_SEPARATOR);
+    AppContext::getSettings()->setValue(EnzymeSettings::CHECKED_SUPPLIERS, value);
+}
+
+void FindEnzymesDialog::sl_updateSuppliers() {
+    const auto& loadedSuppliers = EnzymesSelectorWidget::getLoadedSuppliers();
+    cbSuppliers->clear();
+    cbSuppliers->addItems(loadedSuppliers);
+    QString selStr = AppContext::getSettings()->getValue(EnzymeSettings::CHECKED_SUPPLIERS).toString();
+    static const QString notDefinedTr = EnzymesIO::tr(EnzymesIO::NOT_DEFINED_SIGN);
+    selStr = selStr.replace(EnzymesIO::NOT_DEFINED_SIGN, notDefinedTr);
```

Comment:
Replace the translated value after the split.
This is the last improvement needed. Once test pass I will approve.